### PR TITLE
PCH Title Suggestions: More updates to the UI

### PR DESCRIPTION
**Pull request by vaurdan**
## Description
This PR updates the UI of the Title Suggestions panel to be updated with the latest proposed UI. It also adds a new confirmation modal when applying a title, and improves the loading state.

## Motivation and context
* Improve the UX of the Parse.ly Content Helper and update Related Posts panel accordingly to the proposed UI.
* This PR is based upon #2263 and should be merged **only** after that PR is also merged. It should be automatically rebased to `redesign` when that happens.
* This PR is to be merged into #2238 

## How has this been tested?
Tested locally with all the new changes validated.

### CODE REVIEW

**Comment by vaurdan** [comment]
Not sure why the Node build is failing with `Repo is not clean. Did you forget to commit changes to the /build dir?`
I have all the built files commited. It might be related with the fact of being based on `update/related-posts-ui` branch, so let's see if it build properly after #2263 is merged.

File: src/content-helper/common/settings/types/sidebar-settings.d.ts [hunk 1]
```diff
@@ -41,4 +41,5 @@ export interface TitleSuggestionsSettings {
 	Persona: string;
 	Tone: string;
 	PinnedOpen: boolean;
+	SuggestionsOpen: boolean;
```
Comment (acicovic) [inline src/content-helper/common/settings/types/sidebar-settings.d.ts]: Not entirely sure we should be keeping state for these two. I'd think we'd want everything open every time the panel is being opened. I'm worried that users might fold some section and then think it doesn't work? What are your thoughts?

File: src/content-helper/common/css/common.scss [hunk 2]
```diff
@@ -58,3 +58,62 @@
 		margin: 0 !important;
 	}
 }
+
+.wp-parsely-collapsable-panel {
+	display: flex;
+	flex-direction: column;
+	align-items: flex-start;
+	gap: var(--grid-unit-20, 16px);
```
Comment (acicovic) [inline src/content-helper/common/css/common.scss]: Various missing `to_rem()`s in this file?

File: src/content-helper/editor-sidebar/title-suggestions/title-suggestions.scss [hunk 3]
```diff
@@ -100,56 +104,88 @@
 		display: flex;
 		flex-direction: column;
 		align-items: flex-start;
-		gap: 4px;
+		gap: 8px;
```
Comment (acicovic) [inline src/content-helper/editor-sidebar/title-suggestions/title-suggestions.scss]: Some missing `to_rem()`s in this file? Maybe `gap` doesn't work with `to_rem()`? Seeing this for some other values below as well.

File: src/content-helper/editor-sidebar/title-suggestions/component-suggestions.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+ */
+type TitleSuggestionsProps = {
+	suggestions: Title[];
+	isOpen: boolean;
+	onSettingChange: ( key: keyof TitleSuggestionsSettings, value: string|boolean ) => void;
+	isLoading?: boolean;
+};
+
+/**
+ * Renders the Title Suggestions collapsable panel.
```
Comment (acicovic) [inline src/content-helper/editor-sidebar/title-suggestions/component-suggestions.tsx]: Detail: Prefer using `collapsible`. There's also a class name with that string.

File: src/content-helper/editor-sidebar/title-suggestions/component-suggestions.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+ */
+export const TitleSuggestions = ( {
+	suggestions,
+	isOpen,
+	onSettingChange,
+	isLoading = false,
+}: Readonly<TitleSuggestionsProps> ): JSX.Element => {
+	const [ isCollapsed, setIsCollapsed ] = useState<boolean>( isOpen );
+
+	const toggleCollapse = () => {
```
Comment (acicovic) [inline src/content-helper/editor-sidebar/title-suggestions/component-suggestions.tsx:46]: Adding a function comment would be nice.

File: src/content-helper/editor-sidebar/title-suggestions/component-title-suggestion.tsx [hunk 6]
```diff
@@ -65,7 +88,15 @@ export const TitleSuggestion = (
 	// Flag if the current title has been accepted and applied to the post.
 	const titleInUse = currentPostTitle === props.title.title;
 
-	const onClickApply = async () => {
+	const onClickApply = () => {
```
Comment (acicovic) [inline src/content-helper/editor-sidebar/title-suggestions/component-title-suggestion.tsx:96]: Adding a function comment would be nice, same for `onClickReplace()` and `onClickPin()`.

**Review by acicovic — COMMENTED** [review]
Left some minor comments and a more substantial question.

File: src/content-helper/common/settings/types/sidebar-settings.d.ts (hunk 1 above)
Comment (vaurdan) [reply src/content-helper/common/settings/types/sidebar-settings.d.ts]: Yeah, It's a fair point, and I'm actually inclined to agree with you, I think the user experience would be better with these expanded by default. I'm going to remove those settings.

File: src/content-helper/common/css/common.scss [hunk 7]
```diff
@@ -58,3 +58,66 @@
...
+		&:hover {
+			background-color: transparent;
+		}
+	}
+
+	&.is-opened {
+		padding: 0;
+
+		> div:first-of-type {
+			margin-top: calc(-1 * var(--grid-unit-20, 16px));
```
Comment (acicovic) [inline src/content-helper/common/css/common.scss]: Missing `to_rem()`?

File: src/content-helper/editor-sidebar/title-suggestions/title-suggestions.scss [hunk 8]
```diff
@@ -100,108 +104,124 @@
...
 	}
 
 	// Container for the title suggestions.
-	.wp-parsely-title-suggestions-container {
+	.title-suggestions-container {
 		display: flex;
-		padding: 6px 0 var(--grid-unit-20, 16px) 0;
 		flex-direction: column;
 		align-items: flex-start;
 		gap: var(--grid-unit-20, 16px);
```
Comment (acicovic) [inline src/content-helper/editor-sidebar/title-suggestions/title-suggestions.scss]: Missing `to_rem()`?

File: src/content-helper/editor-sidebar/title-suggestions/title-suggestions.scss [hunk 9]
```diff
@@ -100,108 +104,124 @@
...
-				text-transform: uppercase;
-
-				.components-panel__arrow {
-					right: 0;
-				}
-			}
-		}
+	h2 {
+		color: var(--sidebar-black, #1e1e1e);
+		font-size: 16px;
```
Comment (acicovic) [inline src/content-helper/editor-sidebar/title-suggestions/title-suggestions.scss]: Missing `to_rem()`?

File: src/content-helper/editor-sidebar/title-suggestions/title-suggestions.scss [hunk 10]
```diff
@@ -100,108 +104,124 @@
...
+		font-style: normal;
+		font-weight: 600;
+		line-height: 24px; /* 150% */
```
Comment (acicovic) [inline src/content-helper/editor-sidebar/title-suggestions/title-suggestions.scss]: Missing `to_rem()`?

**Review by acicovic — APPROVED** [review]

**Pull request merged**